app/src/main/java/com/example/ithaca_transit_android_v2/NetworkingAdapters.kt

```diff
@@ -44,4 +46,128 @@ class LocationAdapter {
         }
         return DataLocation(final)
     }
+}
+
+@JsonClass(generateAdapter = true)
+class RouteAdapter {
+    class DataRoute(
+        val data: DataRouteOptions
+    )
+
+    @JsonClass(generateAdapter = true)
+    class DataRouteOptions(
+        val fromStop: List<JsonRoute>,
+        val walking: List<JsonRoute>,
+        val boardingSoon: List<JsonRoute>
+    )
+
+    @JsonClass(generateAdapter = true)
+    class JsonRoute(
+        @Json(name = "directions")
```

Comment:
Backend seems  to also pass us a boundingBox: value. I'm not exactly sure what it does, but I think it might be useful later on when we set the size of the map.

Reply:
Possibly, also add the variable routeID. Right now we don't have a way of identifying each route.

Reply:
i would hold off on adding `boundingBox` until we're sure that we need it for the map. it's better to just add what we know we 100% need at this stage.

Reply:
@JiangoJ also, isn't `routeId` the same as `busId`, no reason to include both unless we need it for the UI

Reply:
So I have no clue what busId is (this doesnt exist on iOS or backend?) but i know `routeId` is a unique identifier for each returned route from backend. This can change per search even though the actual path and route is identical. `tripId` is an identifier for a specific trip that a specific bus number is taking. So for a route that has 3 busses, with numbers `30`, `32`, and `90`, they could (if they have valid live tracking data and thus non null trip identifiers) have 3 different `tripIds` of information useful for UI. Also `tripId` is used for delay information. `routeNumber` is the field that tells us the bus number that is useful for UI. So this will be the `30` or `32` that you will be using in the cards. Please please please talk to either backend or iOS if u have questions like these. We have been using these endpoints for quite a while and there is no reason for you guys to independently do the same research and experimenting that has already been done.